File: Task 1/C++/Components/Product/Digital.cpp
#include "Digital.h"

Digital::Digital() : Games(){
    developer = "";
};

Digital::Digital(std::string name, float cost, std::string supplier, float discount_amount, int packing_units, float cost_per_unit, std::string genre, std::string developer) 
            : Games(name, cost, supplier, discount_amount, packing_units, cost_per_unit, genre){
    this->developer = developer;
}

Digital::Digital(const Digital& other) : Games(other) {
    this->developer = other.developer;
}

std::string Digital::display_product_info(){
    return Games::display_product_info()  
            +"\nDeveloper: "+developer; 
}

File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Product/Digital.java
package com.logistics.Components.Product;

public class Digital extends Games {

    private String developer;

    public Digital() {
        super();
        developer = "";
    }

    public Digital(String name, float cost, String supplier, float discountAmount, int packingUnits, float costPerUnit, String genre, String developer) {
        super(name, cost, supplier, discountAmount, packingUnits, costPerUnit, genre);
        this.developer = developer;
    }

    // Copy Constructor
    public Digital(Digital other) {
        super(other);
        this.developer = other.developer;
    }

    // Clone method
    @Override
    public Digital clone() {
        return new Digital(this);
    }

    // Getters
    public String getDeveloper() { return developer; }

    // Setters
    public void setDeveloper(String developer) { this.developer = developer; }

    @Override
    public String display_product_info() {
        return super.display_product_info() + "\nDeveloper: " + developer;
    }
}
